### PR TITLE
Implement rolling deployments

### LINKED ISSUE #456: Use 'buildpacks' instead of deprecated 'buildpack' in manifest files
Use 'buildpacks' instead of deprecated 'buildpack' in manifest files. https://docs.cloudfoundry.org/devguide/deploy-apps/manifest-attributes.html#buildpack-deprecated

I think `autopilot` needs `buildpack` (see https://github.com/concourse/cf-resource/pull/72#issuecomment-476465049) so this is blocked by exploring/implementing autopilot alternatives (#3597)

### LINKED ISSUE #536: Implement rolling deployments
**What we're after:**
Implement rolling deployments using cf cli version 7 so we can implement CMS/Proxy settings (see https://github.com/fecgov/fec-accounts/issues/289). Change in all repos for consistency/developer experience.

Example: `cf push proxy --strategy rolling -f manifest_dev.yml `

**Completion criteria**
- [x] Read the docs https://docs.cloudfoundry.org/devguide/deploy-apps/rolling-deploy.html
- [x] Switch out zero-downtime-deploy with  --strategy rolling
- [x] Move application manifest attributes from top-level to underneath the app (required for rolling)
- [x] Handle partially failed deployments with `cf cancel-deployment` commands. 
- [x] Consider printing output from https://docs.cloudfoundry.org/devguide/deploy-apps/rolling-deploy.html#status 
- [x] Remove any references to `autopilot`
- [x] Update README

### PR DESCRIPTION
**Pull request by lbeaufort**
Resolves #536 
Resolves #456 

- Reformatted manifests with newer format
- Implemented rolling deployments
- Moved start command to `Procfile` to match other apps, and also because there seems to be a bug where the start command isn't recognized when it's specified in the manifest file. See https://github.com/cloudfoundry/cloud_controller_ng/issues/1922 and https://app.circleci.com/pipelines/github/fecgov/fec-eregs/152/workflows/ee83ec7e-aec1-4555-9573-515696ab86fe/jobs/463 

- ~Because we're deploying in place~ we're always (got it on `develop`) going to get a message like:
```
 Found another file with the destination path 'regulations/js/unittests/specs/views/comment-confirm-view-spec.js'. It will be ignored since only the first encountered file is collected. If this is not what you want, make sure every static file has a unique path.
```
I tried to move the `collectstatic` with `--clear` to the `run.sh` command with the env var `DISABLE_COLLECSTATIC=1` like the CMS, but the styles wouldn't compile that way. I confirmed by modifying `fec.js` and `site-nav.js` that the newer JS files will still take precedence, so we can ignore those warnings.

We ssh'ed into the app after multiple rolling deployments, went to `/app/compiled/regulations/` and only found one (recent) copy of the static assets there.

## Reviewers
- At least two reviewers please

## How to test the changes locally

**Through CircleCI**
- Make your own copy of this branch (please don't commit to this branch)
- Modify deploy task to deploy to a space of your choice
- Push to Github, make sure build works

**Locally:**
- Install cf cli version [following this guide](https://github.com/cloudfoundry/cli/wiki/Version-Switching-Guide) or:
```
# Install cf-cli@7. Complete commands
brew install cf-cli@7

# If you run across an error like this: Error: No such keg: /usr/local/Cellar/cf-cli@6, make sure you install cf-cli version 6 with brew
brew install cf-cli@6
 
# Unlink cf-cli version 6 and link cf-cli version 7
brew unlink cf-cli@6 && brew unlink cf-cli@7 && brew link cf-cli@7 && cf version
```
- Do a manual deploy with `invoke deploy --space dev`

##  Testing failed builds
- Test with bad requirements.txt, like `Django=2222`, make sure build fails
- Test with crashing app - add 5 instances,  at the top of `run.sh`, add:
```
rand_mod=$(expr $RANDOM % 5)
if [ $rand_mod -eq 0 ]; then
    echo "Starting divisible by five app"
else
    echo "Crashing non-divisible by five app";
    exit 1
fi
```
make sure build fails and is cancelled

## Impacted areas of the application
List general components of the application that this PR will affect:

-  Deploy task only

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -208,5 +208,6 @@ Environment | URL                              | Proxy | Description
 $ pip install -r requirements.txt   # updates the -core/-site repositories
 $ npm run build
 $ python manage.py compile_frontend   # builds the frontend
-$ cf target -s ${cf_space} && cf zero-downtime-deploy eregs -f manifest.${cf_space}.yml
+$ cf target -s ${cf_space}
+$ cf push --strategy rolling eregs -f manifest.${cf_space}.yml
```
Comment (lbeaufort) [inline README.md:212]: swap this out with `invoke deploy` command?

**Review by patphongs — APPROVED** [review]
Looks good, thanks @lbeaufort !

**Review by jason-upchurch — APPROVED** [review]
Looks great @lbeaufort ! Thank you for walking us through this!

**Review by fec-jli — APPROVED** [review]
good

**Pull request merged**